### PR TITLE
(678) Email team leads on project create

### PR DESCRIPTION
**Pull request by jkempster34**
### Update Sidekiq to the latest version
We are seeing a warning message when running Sidekiq that has been fixed in
the latest version.

### Add GOV.UK Notify mailer Gem
Add the `mail-notify` Gem in order to integrate Notify with ActionMailer.

### Send email to all team leads when a project is created
We want to alert users with the team lead role when a new project has been
created.

### Don't use Sidekiq as the ActiveJob queue adapter in all environments
Currently, the app is configured to use Sidekiq as the ActiveJob queue adapter
in all environments.

For simplicity, we can use the default `async` queue adapter in development
because it doesn't need any external infrastructure.

We will need to configure the queue adapter as Sidekiq for production.

image

## Checklist

- [x] Attach this pull request to the appropriate card in Trello.
- [x] Update the `CHANGELOG.md` if needed.

### CODE REVIEW

**Comment by jkempster34** [comment]
[System can send an email notification](https://trello.com/c/LbDgPRaa/678-system-can-send-an-email-notification)

File: app/mailers/team_leader_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class TeamLeaderMailer < ApplicationMailer
+  def new_project_created(team_leader, project)
+    template_mail(
+      "ea4f72e4-f5bb-4b1a-b5f9-a94cc1840353",
```
Comment (mec) [inline app/mailers/team_leader_mailer.rb:4]: Hmm, will all the template IDs get unwieldly as we add more? Should they be environment variables?

**Review by mec — CHANGES_REQUESTED** [review]
This is looking great! It's very cool we can get so much value out of so little change! 

I could approve this - but I am interested in your thoughts on my comments so I am setting request changes.

File: Gemfile.lock [hunk 2]
```diff
@@ -321,10 +321,10 @@ GEM
       concurrent-ruby (~> 1.0, >= 1.0.2)
     shoulda-matchers (5.1.0)
       activesupport (>= 5.2.0)
-    sidekiq (6.5.5)
```
[the remark this answers is not in the document]
Comment (jkempster34) [reply Gemfile.lock:324]: It was just a deprecation warning filling up my terminal 

```
Passing the timeout as a positional argument is deprecated, it should be passed as a keyword argument sidekiq
```

File: app/controllers/projects_controller.rb [hunk 3]
```diff
@@ -28,13 +28,20 @@ def create
...
       TaskListCreator.new.call(@project, workflow_root: DEFAULT_WORKFLOW_ROOT)
+      notify_team_leaders
 
       redirect_to project_path(@project), notice: I18n.t("project.create.success")
     else
       render :new
     end
   end
 
+  private def notify_team_leaders
```
[the remark this answers is not in the document]
Comment (jkempster34) [reply app/controllers/projects_controller.rb:39]: I reckon the TaskListCreator should be responsible for generating the task list only. If you look at the TaskListCreator, its currently far more complicated that the ProjectsController.

[the remark this answers is not in the document]
Comment (mec) [reply app/controllers/projects_controller.rb:39]: Keeping controllers thin and the complexity out is, generally speaking, a good thing. I agree the TaskListCreator is getting bloated, but it is responsible for creating a new task list and that is the point at which the email is sent.

One benefit of not having this in the controller is that it should make testing it simpler.

Happy to keep this and see where we go once more notification come along.

File: app/mailers/team_leader_mailer.rb (hunk 1 above)
Comment (mec) [reply app/mailers/team_leader_mailer.rb:4]: I think this is good, just thinking ahead and about how best to handle these IDs, this one is encapsulated in the `TeamLeaderMailer` which I like, but I can see this coming apart when we have to send to multiple roles...

However we tackle this one sets the tone for those that follow (in my experience) so this is our chance to lay some best practice foundaitons.

**Review by mec — APPROVED** [review]
🚢

**Pull request merged**